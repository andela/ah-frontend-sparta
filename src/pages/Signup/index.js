import React from 'react';
import SignupForm from '../../components/SignupForm';
import { FaceBook, Google, Twitter } from '../../assets/icons'
import logo from '../../assets/images/logo.png';
import '../../assets/scss/Login.scss';

class LoginPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      signinStateButton: 'active',
      signupStateButton: 'inactive'
    };
  }

  onSigninToggle() {
    this.setState({
      signinStateButton: 'active',
      signupStateButton: 'inactive'
    });
  }

  onSignupToggle() {
    this.setState({
      signinStateButton: 'inactive',
      signupStateButton: 'active'
    });
  }

  render() {

    return (
      <div className="outer">
        <div className="innerHalf1">
          <img src={logo} alt="Logo" className="logo" />
          <div className="intro">
            We create a community of like minded authors to foster inspiration
            and innovation by leveraging the modern web.
          </div>
        </div>
        <div className="innerHalf2">
          <div>
            <div className="innerHalfContent">
              <div className="socialDiv">
                Continue with {' '}
                <FaceBook />{' '}
                <Twitter />{' '}
                <Google />
              </div>
              <div className="signStr">Or</div>
              <SignupForm />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default LoginPage;
