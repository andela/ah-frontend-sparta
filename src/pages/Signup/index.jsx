import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import SignupForm from '../../components/SignupForm';
import { FaceBook, Google, Twitter } from '../../assets/icons';
import logo from '../../assets/images/logo.png';
import '../../assets/scss/Login.scss';
import { signUpUser } from '../../actions/signupSigninActions';

export class Signup extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      username: '',
      password: '',
      confirmPassword: '',
      error: '',
    };
  }

  onChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { registerCurrentUser, history } = this.props;
    const {
      username, email, password, confirmPassword,
    } = this.state;
    if (password !== confirmPassword) {
      this.setState({
        error: 'Your passwords do not match',
      });
    } else if (
      username === ''
      || email === ''
      || password === ''
      || confirmPassword === ''
    ) {
      this.setState({
        error: 'Please populate all the fields',
      });
    } else {
      const signupData = {
        user: {
          email,
          password,
          username,
        },
      };
      registerCurrentUser(signupData, history);
    }
  };

  render() {
    const { error } = this.state;
    return (
      <div className="outer">
        <div className="innerHalf1">
          <img src={logo} alt="Logo" className="logo" />
          <div className="intro">
            We create a community of like minded authors to foster inspiration
            and innovation by leveraging the modern web.
          </div>
        </div>
        <div className="innerHalf2">
          <div>
            <div className="innerHalfContent">
              <div className="socialDiv">
                Continue with
                {' '}
                <FaceBook />
                {' '}
                <Twitter />
                {' '}
                <Google />
              </div>
              <div className="signStr">Or</div>
              <SignupForm
                error={error}
                handleSubmit={this.handleSubmit}
                onChange={this.onChange}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Signup.propTypes = {
  registerCurrentUser: PropTypes.func,
  history: PropTypes.func,
};

Signup.defaultProps = {
  registerCurrentUser: () => {},
  history: () => {},
};

export default withRouter(
  connect(
    null,
    { registerCurrentUser: signUpUser },
  )(Signup),
);
