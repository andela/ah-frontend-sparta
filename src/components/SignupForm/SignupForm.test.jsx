import React from 'react';
import renderer from 'react-test-renderer';
import Signup from './index';

jest.mock('react-router-dom');

test('snapshot test', () => {
  const wrapper = renderer.create(<Signup />);
  expect(wrapper).toMatchSnapshot();
});
