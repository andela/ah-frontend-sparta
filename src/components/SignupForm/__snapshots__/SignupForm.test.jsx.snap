// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div>
  <div
    className="signStr spacebottom"
  >
    Create an account
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="inputContain"
    >
      <input
        className="input"
        id="loginInput"
        name="username"
        onChange={[Function]}
        placeholder="USERNAME"
        type="text"
        value=""
      />
    </div>
    <div
      className="inputContain"
    >
      <input
        className="input"
        id="loginInput"
        name="email"
        onChange={[Function]}
        placeholder="EMAIL"
        type="email"
        value=""
      />
    </div>
    <div
      className="inputContain"
    >
      <input
        className="input"
        id="loginInput"
        name="password"
        onChange={[Function]}
        placeholder="PASSWORD"
        type="password"
        value=""
      />
    </div>
    <div
      className="inputContain"
    >
      <input
        className="input"
        id="loginInput"
        name="confirmPassword"
        onChange={[Function]}
        placeholder="CONFIRM PASSWORD"
        type="password"
        value=""
      />
    </div>
    
    <button
      id="loginButton"
      onClick={[Function]}
      type="submit"
    >
      Signup
    </button>
    <div
      className="other"
    >
      <span>
        Already have an account?
         
      </span>
    </div>
  </form>
</div>
`;
